src/key.cpp

```diff
@@ -332,6 +334,40 @@ bool CKey::Derive(CKey& keyChild, ChainCode &ccChild, unsigned int nChild, const
     return ret;
 }
 
+// Returns just the x-coordinate in big endian
+static int bip324_ecdh_hash(unsigned char* output, const unsigned char* x32, const unsigned char* y32, void* data)
+{
+    memcpy(output, x32, 32);
+    return 1;
+}
+
+bool CKey::ComputeBIP324ECDHSecret(const CPubKey& pubkey, const Span<uint8_t> initiator_hdata, const Span<uint8_t> responder_hdata,
```

Comment:
You can make the arguments here take `Span<const uint8_t>` - or even `Span<const std::byte>`. I don't think this function needs to be able to mutate the passed in hdata.